rule_stop.go

```diff
@@ -1,12 +1,19 @@
 package techan
 
-import "github.com/sdcoffey/big"
+import (
+	"github.com/sdcoffey/big"
+)
 
 type stopLossRule struct {
 	Indicator
 	tolerance big.Decimal
 }
 
+type stopGainRule struct {
```

Comment:
You actually don't need to define a duplicate struct here, since this one is private. I would redefine the above rule as 

```go
type stopRule struct {
    Indicator
    tolerace big.Decimal
}
```

and then make an alias for both `stopLoss` rule and `stopGain` rule, eg.

```go
type stopLossRule = stopRule
type stopGainRule = stopRule
```